test/spec/services/applicationGeneratorSpec.js

```diff
@@ -3,13 +3,23 @@
 describe("ApplicationGenerator", function(){
   var ApplicationGenerator;
   var inputTemplate;
+  var mathFloor = Math.floor;
 
   beforeEach(function(){
 
     inject(function(_ApplicationGenerator_){
       ApplicationGenerator = _ApplicationGenerator_;
-      ApplicationGenerator._generateSecret = function(){
```

Comment:
Yup. Since the end of the file does this:

```javascript
return {
  // "exporting" the functions...
  _generateSecret: generateSecret
}
```
Doing `ApplicationGenerator._generateSecret = function(){ ... }` is not effective.  The original function is used internally a number of times: 

Example:
```javascript 
// line 233, this call cannot be mocked
secret: generateSecret()  
``` 
So will never use the mocked function.